src/oqsconfig.h.cmake

```diff
@@ -22,6 +22,8 @@
 #cmakedefine OQS_USE_AES_OPENSSL 1
 #cmakedefine OQS_USE_SHA2_OPENSSL 1
 #cmakedefine OQS_USE_SHA3_OPENSSL 1
+#cmakedefine OQS_DLOPEN_OPENSSL 1
+#cmakedefine OQS_OPENSSL_CRYPTO_SONAME "@OQS_OPENSSL_CRYPTO_SONAME@"
```

Comment:
Just for my personal understanding: Where is this new config variable "OQS_OPENSSL_CRYPTO_SONAME" needed?

Reply:
This is used as the first argument of `dlopen`.

Reply:
Ah, thanks for the pointer. That then begs a followup question: This variable is gleaned at build time (part of `cmake` execution), right? Does this mean in turn that this facility will break when a binary containing `liboqs` built with this option is run on another machine (where the .so is located somewhere else)?

Reply:
As the variable doesn't contain path, the lookup logic is up to the dynamic linker (ld.so) on another machine. For example, on Fedora 39 the variable would end up with "libcrypto.so.3", where the corresponding shared library resides in /usr/lib64/ on x86_64. If you copy liboqs built with this dlopen feature into another distribution, such as Debian with OpenSSL 3.0, it should still be able to detect the shared library even if the location is /usr/lib/x86_64-linux-gnu/.

On the other hand, it is in theory possible to support arbitrary shared library of OpenSSL, through dlopen, as we [do](https://github.com/golang-fips/openssl/blob/576fe0d377882f8d0fd6537762ef2ff7918facc8/openssl_test.go#L22) in OpenSSL binding in Go, though this is not our current motivation behind this PR.

Reply:
> though this is not our current motivation behind this PR.

and it needn't be. Not encoding the path is good (enough). Thanks for the explanation.